src/controllers/selectedAccount/selectedAccount.ts

```diff
@@ -69,10 +135,130 @@ export class SelectedAccountController extends EventEmitter {
     this.emitUpdate()
   }
 
+  async #updateSelectedAccountPortfolio(skipUpdate?: boolean) {
+    if (!this.#portfolio || !this.#defiPositions || !this.account) return
+
+    const now = Date.now()
+    const timeSinceLastCall = now - this.#updatePortfolioThrottle.time
+    if (timeSinceLastCall <= 150) {
+      if (!this.#updatePortfolioThrottle.throttled) {
+        this.#updatePortfolioThrottle.throttled = true
+        await wait(150 - timeSinceLastCall)
+        this.#updatePortfolioThrottle.throttled = false
+        await this.#updateSelectedAccountPortfolio()
+      }
+      return
+    }
+
+    this.#updatePortfolioThrottle.time = now
+
+    const defiPositionsAccountState = this.#defiPositions.state[this.account.addr]
+
+    const updatedPortfolioState = getSelectedAccountPortfolio(
+      structuredClone({
+        latest: this.#portfolio.latest,
+        pending: this.#portfolio.pending
+      }),
```

Comment:
Why do this here? Isn't it better to pass the original object to the function and clone it in the function body?
So, instead of spreading large objects:
```
  const updatedPortfolioState = {
    latest: {
      ...portfolioState.latest,
      [account.addr]: portfolioLatestAccountState
    },
    pending: {
      ...portfolioState.pending,
      [account.addr]: portfolioPendingAccountState
    }
  }
```
we can do:
```
  portfolioState.latest[account.addr] = portfolioLatestAccountState
  portfolioState.pending[account.addr] = portfolioPendingAccountState

return portfolioState
```

Reply:
we should not modify the original state. I tried this pattern portfolioState.latest[account.addr] but for some reason assigning it this way to a structuredClone object was not working in the tests and this is why I ended up spreading it